### DISCUSSION TITLE
Official Allure Plugin for Azure DevOps without downloading directly from github.com

### DISCUSSION THREAD

**Discussion by mhanxchu** [allure-framework/allure2#2450] [Questions & Support] [answered]
I am integrating the official plugin for azure devops pipeline to display it in the pipeline run tab. Currently I am using a selfhosted build agent which is installed in a linux Azure VM. The plugin could find allure test results but there are firewalls issue when the plugin trying to download source from github.com. Are there alternatives to that? Can I manually download the source and make the plugin use that instead or some internal artifactory platform, because I would like to keep the deps internally.

**Comment by epszaw** [comment]
I think, it's possible to provide the ability to do that 👍

**Reply by epszaw** [reply]
Published a new version (`1.3.0`) which supports configuring custom allure report download path.

To download the report from a custom url add `allureDownloadUrl` (default value is `https://github.com/allure-framework/allure2/releases/download/{{allureVersion}}/allure-{{allureVersion}}.tgz`, where `allureVersion` will be replaced by `allureVersion` input value).

Please keep in mind, that the file should be a `.tar` archive!

**Reply by mhanxchu** [reply]
@epszaw Thanks a lot for the fast support! So I just need to specify e.g. the artifactory url path where the tgz file is kept right? How does the download work, as for Artifactory, I need to provide credentials to access that.

**Reply by mhanxchu** [reply]
I tried to test it with putting the tar archive file in our Azure Repo and put an url to that as allureDownloadUrl but I received the following error

Error: Unexpected HTTP response: 203
    at /mnt/_buildkit/_tasks/PublishAllureReport_867c4cd2-7a10-11ed-a1eb-0242ac120189/1.3.0/node_modules/azure-pipelines-tool-lib/tool.js:216:31
    at Generator.next (<anonymous>)
    at fulfilled (/mnt/_buildkit/_tasks/PublishAllureReport_867c4cd2-7a10-11ed-a1eb-0242ac120189/1.3.0/node_modules/azure-pipelines-tool-lib/tool.js:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  httpStatusCode: 203

What does this mean?

**Reply by epszaw** [reply]
It means that we need to provide some authentication data to the request. Would it be possible to configure artifactory to respond all the requests from azure host without username/password pair?

**Comment by baev** [comment, marked as answer]
The latest version (1.4.0) of the plugin now includes a bundled Allure Commandline. To use the bundled version, you can omit the version from the configuration.
